pkg/mutation/system.go

```diff
@@ -107,12 +108,20 @@ func (s *System) Mutate(obj *unstructured.Unstructured, ns *corev1.Namespace) (b
 			}
 			if cmp.Equal(original, obj) {
 				if *MutationLoggingEnabled {
-					logAppliedMutations("Oscillating mutation.", original, allAppliedMutations)
+					logAppliedMutations("Oscillating mutation.", original, allAppliedMutations, nil)
 				}
 				return false, fmt.Errorf("oscillating mutation for %s %s %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind, obj.GetNamespace(), obj.GetName())
 			}
+			mutationTimestamp := time.Now()
 			if *MutationLoggingEnabled {
-				logAppliedMutations("Mutation applied", original, allAppliedMutations)
+				logAppliedMutations("Mutation applied", original, allAppliedMutations, &mutationTimestamp)
+			}
+
+			if *MutationAnnotationsEnabled {
+				err := mutationAnnotations(obj, allAppliedMutations, mutationTimestamp)
```

Comment:
allAppliedMutations will be nil unless logging is enabled. This couples the two command line flags in a weird way.

If we are only reporting the set of annotations on the resource, maybe we should manage the creation of the set here?

Reply:
fixed, we're now collecting when either logging or annotating is enabled